data-source-manager/src/main/java/com/odysseusinc/datasourcemanager/krblogin/KrbConfig.java

```diff
@@ -51,8 +53,16 @@
         if (kinitCommand == null) {
             kinitParamsLine = "";
         } else {
-            String[] kinitParams = Arrays.copyOfRange(getKinitCommand(), 1, getKinitCommand().length);
-            kinitParamsLine = StringUtils.join(kinitParams, " ").replace(keytabPath.toString(), KRB_KEYTAB_PATH);
+            List<String> kinitParamsList = Arrays.asList(kinitCommand);
```

Comment:
The block of code is very hard to read and understand. Pls try to rewrite it or put comprehensive comments